rfcs/xxxx-custom-event-metadata-from-annotations/README.md

```diff
@@ -0,0 +1,184 @@
+
+## Motivation
+
+This RFC comes as a response to the need for adding custom metadata to events about Flux objects
+sent to notification providers. See specific user stories in the [User Stories](#user-stories) section.
+
+### Goals
+
+Provide a method for Flux users to embed custom/static metadata in their Flux objects
+and have that metadata propagated to the notification providers.
+
+### Non-Goals
+
+In this proposal we **do not** aim to provide a method for Flux users to send etcd-indexed custom metadata
+fields from Flux objects in events to notification-controller, most specifically labels. By design an event
+already contains enough identification information to locate the associated Flux object inside the cluster,
+which covers the use case of labels. Flux does not wish to incentivize practices that are impactful to clusters
+without a strong reason or benefit.
+
+## Proposal
+
+When sending events about Flux objects, we propose sending annotation keys prefixed with a well-defined
+API Group in addition to those already sent by the controller.
+
+### User Stories
+
+#### Story 1
+
+> As a user, I want to embed FluxCD into my GitHub Workflow in a way that it only succeeds if
+> the deployment made by FluxCD is successful.
+
+For example, embedding a Deployment ID from the GitHub API in a `HelmRelease` object like the one below:
+
+```yaml
+apiVersion: helm.toolkit.fluxcd.io/v2
+kind: HelmRelease
+metadata:
+  name: podinfo
+  namespace: flux-system
+  annotations:
+    event.toolkit.fluxcd.io/deploymentID: e076e315-5a48-41c3-81c8-8d8bdee7d74d
+spec:
+  chart:
+    spec:
+      chart: podinfo
+      version: 6.5.*
+      sourceRef:
+        kind: HelmRepository
+        name: podinfo
+```
+
+Should cause notification-controller to propagate an event like the one below (most fields omitted for brevity):
+
+```json
+{
+  "involvedObject": {
+    "apiVersion": "helm.toolkit.fluxcd.io/v2",
+    "kind": "HelmRelease",
+    "name": "podinfo",
+    "namespace": "flux-system",
+    "uid": "7d0cdc51-ddcf-4743-b223-83ca5c699632"
+  },
+  "metadata": {
+    "deploymentID": "e076e315-5a48-41c3-81c8-8d8bdee7d74d"
+  }
+}
+```
+
+#### Story 2
+
+> As a user, I want to embed the new image tag in a `HelmRelease` object when the image is updated by an `ImageUpdateAutomation`
+> and have that information propagated to the notification providers.
+
+For example:
+
+```yaml
+apiVersion: helm.toolkit.fluxcd.io/v2
+kind: HelmRelease
+metadata:
+  name: podinfo
+  namespace: flux-system
+  annotations:
+    event.toolkit.fluxcd.io/image: ghcr.io/stefanprodan/podinfo # {"$imagepolicy": "flux-system:podinfo:name"}
+    event.toolkit.fluxcd.io/imageTag: 6.5.0 # {"$imagepolicy": "flux-system:podinfo:tag"}
+spec:
+  chart:
+    spec:
+      chart: podinfo
+      version: 6.5.0 # {"$imagepolicy": "flux-system:podinfo:tag"}
+      sourceRef:
+        kind: HelmRepository
+        name: podinfo
```

Comment:
```suggestion
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: podinfo
  namespace: flux-system
  annotations:
    event.toolkit.fluxcd.io/image: latest # {"$imagepolicy": "flux-system:podinfo"}
spec:
  chart:
    spec:
      chart: podinfo
      sourceRef:
        kind: HelmRepository
        name: podinfo
      values: 
        image:
          tag: latest  # {"$imagepolicy": "flux-system:podinfo:tag"}
```

Reply:
But the image tag is not yet accessible in event annotation yet?

Reply:
Image automation will write these to Git, this how Flux works.

Reply:
@stefanprodan shouldn't `.values` be under `.spec`?

Reply:
Yes, wrong indentation in GH UI.